src/source/geojson_source.js

```diff
@@ -177,6 +177,8 @@ class GeoJSONSource extends Evented implements Source {
      * });
      */
     setData(data: GeoJSON | string) {
+        // set fade duration to 0 to prevent fading out of symbols
+        if (this.map && this._loaded) this.map._fadeDuration = 0;
```

Comment:
Since this sets the global state, calling `setData` once will disable fading forever, even though we only needed it for this one source update, right?